File: src/test/java/kitchenpos/application/fixture/OrderTableFixtures.java
package kitchenpos.application.fixture;

import java.util.UUID;
import kitchenpos.domain.OrderTable;

public final class OrderTableFixtures {

    private OrderTableFixtures() {
        throw new RuntimeException("생성할 수 없는 클래스");
    }

    public static OrderTable createOrderTable(
        UUID id,
        String name,
        int numberOfGuests,
        boolean empty
    ) {
        final OrderTable orderTable = new OrderTable();
        orderTable.setId(id);
        orderTable.setName(name);
        orderTable.setNumberOfGuests(numberOfGuests);
        orderTable.setEmpty(empty);
        return orderTable;
    }

    public static OrderTable createOrderTable(
        String name,
        int numberOfGuests,
        boolean empty
    ) {
        return createOrderTable(
            UUID.randomUUID(),
            name,
            numberOfGuests,
            empty
        );
    }
}

File: src/testFixtures/kotlin/domain/OrderTableFixtures.kt
package domain

import kitchenpos.domain.OrderTable
import java.util.*

object OrderTableFixtures {
    fun makeOrderTableOne(): OrderTable {
        return OrderTable().apply {
            this.id = UUID.fromString("6924640b-b0fc-4c86-84f9-b750eeba0205")
            this.name = "테이블 1"
            this.numberOfGuests = 4
            this.isOccupied = true
        }
    }

    // 초기 order table
    fun makeOrderTableTwo(): OrderTable {
        return OrderTable().apply {
            this.id = UUID.fromString("6924640b-b0fc-4c86-84f9-b750eeba0205")
            this.name = "테이블 1"
            this.numberOfGuests = 0
            this.isOccupied = false
        }
    }
}
